tests/phpunit/integration/Core/Util/User_Input_SettingsTest.php

```diff
@@ -0,0 +1,193 @@
+<?php
+/**
+ * Class Google\Site_Kit\Tests\Core\Util\User_Input_SettingsTest
+ *
+ * @package   Google\Site_Kit\Tests\Core\Util
+ * @copyright 2020 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://sitekit.withgoogle.com
+ */
+
+namespace Google\Site_Kit\Tests\Core\Util;
+
+use Google\Site_Kit\Context;
+use Google\Site_Kit\Core\Storage\User_Options;
+use Google\Site_Kit\Core\Util\User_Input_Settings;
+use Google\Site_Kit\Tests\Core\Util\FakeUser_Input_Settings;
+use Google\Site_Kit\Tests\TestCase;
+
+class User_Input_SettingsTest extends TestCase {
+
+	/**
+	 * Context object.
+	 *
+	 * @var Context
+	 */
+	private $context;
+
+	/**
+	 * @var bool
+	 */
+	protected static $old_wp_using_ext_object_cache;
+
+	public static function wpSetUpBeforeClass() {
+		self::$old_wp_using_ext_object_cache = wp_using_ext_object_cache();
+	}
+
+	public static function wpTearDownAfterClass() {
+		wp_using_ext_object_cache( self::$old_wp_using_ext_object_cache );
+	}
+
+	public function setUp() {
+		parent::setUp();
+		$this->context = new Context( GOOGLESITEKIT_PLUGIN_MAIN_FILE );
+	}
+
+	public function test_not_connected_to_proxy() {
+		$settings = new User_Input_Settings( $this->context );
+		$results  = array(
+			$settings->get_settings(),
+			$settings->set_settings( array() ),
+		);
+
+		foreach ( $results as $result ) {
+			$this->assertTrue( is_wp_error( $result ) );
```

Comment:
The base WP TestCase has assertions for this 🙂 
```suggestion
			$this->assertWPError( $result );
```